subworkflows/local/generate_downstream_samplesheet.nf

```diff
@@ -103,10 +104,35 @@ workflow GENERATE_DOWNSTREAM_SAMPLESHEET {
                     .set { ch_mag_metadata }
             }
 
-            // Create samplesheet for each sample using meta information
-            ch_mag_id_samplesheets = ch_mag_metadata.collectFile() { meta ->
-                [ "${meta.id}_phageannotator_samplesheet.csv", "sample,group,fastq_1,fastq_2,fasta" + '\n' + "${meta.id},${meta.group},${meta.fastq_1},${meta.fastq_2},${meta.fasta}" + '\n' ]
-            }
+                // Create samplesheet for each sample using meta information
+                ch_mag_id_samplesheets = ch_mag_metadata.collectFile() { meta ->
+                    // Save reads and assemblies to outdir so that they are in a stable location
+                    file(meta.fastq_1.toUriString(), checkIfExists: true).copyTo("${params.outdir}/downstream_samplesheets/fastq/${meta.fastq_1.name}")
+                    file(meta.fasta, checkIfExists: true).copyTo("${params.outdir}/downstream_samplesheets/fasta/${meta.fasta.name}")
```

Comment:
Asking for some advise about this, this is indeed very clean, but given it's outside a process/publishdir I'm a bit unsure. E.g. some people use symlinks from work to results dir rather than copy, so this would violate this.

Secondly this woudl potentially result in two copies of the same read files if one of the `--save_*` parameters are given.

It maybe we have to come up with some complicated logic that instead picks the right 'final' directory for the proceessing reads, and it's that that you append to the beginning, and then add.

I think this would come with two steps:

1. Make sure one of `--save_clipped_reads`, `--save_hostremoved_reads`, `--save_phixremoved_reads`, `--save_bbnorm_reads` are selected if `params.generate_samplesheet` is true

    So some input validation code like:
    ```
    if ( params.generate_samplesheet && ![params.saved_clipped, save.hostremoved <...>].any() ) { Nextflow.error('[nf-core/mag] ERROR: must at least save one XYX if --generate_samplesheet <...>') }
     ```
     
 2.  Then have to have some complicated logic to select which directory gets appended to the beginning of the file name (probably best in a case when but w/e), e.g.
 
     ```
     if (params.save_clipped_reads && !params.save_phix && !params.save_host_removed && !params._save_bbnorm) { 
    samplehseet_reads_dir = /subdir/clipedreads/saved/in
       } else if ( params.save_phix && !params.save_host_removed && !params._save_bbnorm){
       samplesheet_reads_dir = /subdir/phixreads/saved/in
       }
     ```

Reply:
Totally see what you're saying and was curious what thoughts would be about doing this outside a process. I am more than happy change it to what you're suggesting!

The reason I was giving this approach a try was I was hoping to find a generalizable approach that would work across pipelines/updates even if the save input logic you're talking about gets modified (also, I selfishly wanted to find a quicker approach haha) Also, I was hoping to force a copy because in the past I've symlinked to publishDir thinking I copied and had my reads disappear because my workDir is in a scratch directory 😅.

Maybe we could make a module that just collects/copies the files, then at least were getting inputs and outputs checked for existence and it would take care of issue #1 with this approach.

Still, I definitely have though about what you're saying and will make that change if you think it's best!